feat: pokedex api pokemon detail

NicoEnriquezFFuFm opened the pull request:
- added version_group_detail
- added pokemon_stats
- added pokemon_moves and move
- added pokemon pokemondetail_abilities
- added pokemon pokemondetail_ability
- added move_learn_method
- added pokemon_detail
- added a function that displays specific details about pokemon
- removed additonal functions that displays one specific data

![image](https://user-images.githubusercontent.com/128012634/226539330-c25ee923-714d-4cb4-b6f4-892a790680c1.png)

NicoEnriquezFFuFm committed:
feat: added additional freezed models
- added version_group_detail
- added pokemon_stats
- added pokemon_moves and move
- added pokemon pokemondetail_abilities
- added pokemon pokemondetail_ability
- added move_learn_method
- added pokemon_detail

NicoEnriquezFFuFm committed:
feat: added additional freezed models
- added a function that displays specific details about pokemon
- removed additonal functions that displays one specific data
- modified the variable placement(pokemons) inside main.dart
- modified to list for sub models in pokemon_detail.dart
- modified to list for sub models in pokemon_moves.dart

NicoEnriquezFFuFm committed:
feat: added addtional freezed models
- merged conflicts with feature/pokedex_api_get_pokemon_list

NicoEnriquezFFuFm committed:
feat: added additional freezed models
- enhanced structure of pokemon_detail.dart model
- enhanced structure of pokemon_moves.dart model
- enhanced structure of pokemon_api.dart

NicoEnriquezFFuFm committed:
feat: added additional freezed models
- merged conflicts with review

NicoEnriquezFFuFm committed:
feat: added additional freezed models
- merge conflicts with review on main.dart

NicoEnriquezFFuFm committed:
feat: added additional freezed models
- merged conflicts on pokemon_api.dart

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -12,19 +15,13 @@ class PokemonApi {
-    });
+    Response response = await apiClient.dio.getUri(uri);
+    PokemonDetail pokeResponse = PokemonDetail.fromJson(response.data);
+    return pokeResponse;
```
just return immediately the `PokemonDetail.fromJson(response.data);` instead

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -12,19 +15,13 @@ class PokemonApi {
-    queryParam['limit'] = limit;
-    queryParam['offset'] = offset;
-
+  Future<PokemonDetail> getPokemonDetails({required String id}) async {
```
use pokemon's name instead of id 🙂

shawn-ffufm reviewed (changes requested):
@NicoEnriquezFFuFm see minor feedback

NicoEnriquezFFuFm committed:
feat: added additional freezed models
- modified function getPokemonDetails parameters from id to name

- modified getPokemonDetails to return directly from the PokemonDetail.fromJson()

NicoEnriquezFFuFm committed:
feat: added addtional freezed models
- removed extra line on pokemon_detail.dart

NicoEnriquezFFuFm committed:
feat: added additional freezed models
- (pr feedback)
-  - added the getPokemonList function that gets the list of pokemon
